plugins/formal-verification/src/org/jetbrains/kotlin/formver/scala/silicon/ast/Domain.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.scala.silicon.ast
+
+import org.jetbrains.kotlin.formver.conversion.ConvertedType
+import org.jetbrains.kotlin.formver.conversion.ConvertedVar
+import org.jetbrains.kotlin.formver.scala.IntoViper
+import org.jetbrains.kotlin.formver.scala.emptySeq
+import org.jetbrains.kotlin.formver.scala.toScalaOption
+import org.jetbrains.kotlin.formver.scala.toScalaSeq
+import viper.silver.ast.NamedDomainAxiom
+
+// Cannot implement `IntoViper` as we need to pass the domain name.
+class DomainFunc(
+    val name: String,
+    val formalArgs: List<ConvertedVar>,
+    val typ: ConvertedType,
+    val unique: Boolean,
+    val pos: Position = Position.NoPosition,
+    val info: Info = Info.NoInfo,
+    val trafos: Trafos = Trafos.NoTrafos,
+) {
+    fun toViper(domainName: String): viper.silver.ast.DomainFunc =
+        viper.silver.ast.DomainFunc(
+            name,
+            formalArgs.map { it.toLocalVarDecl() }.toScalaSeq(),
+            typ.viperType.toViper(),
+            unique,
+            null.toScalaOption(),
+            pos.toViper(),
+            info.toViper(),
+            domainName,
+            trafos.toViper()
+        )
+}
+
+// Cannot implement `IntoViper` as we need to pass the domain name.
+class DomainAxiom(
+    val name: String,
```

Comment:
Let's make the name optional and in the case name it is null create an `AnonymousDomainAxiom`. In that way we don't always have to generate a name if we don't want to.

Reply:
Good point, done.